### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by brson**
This is mostly just internal refactoring.

I reorganized the crates into `multirust-cli`, `multirust` (the main engine), `multirust-dist` (dist server and install), and `multirust-mock`, and put them in directories under `src`.

Other cleanup included hiding the tty stuff in its own module, removing the unused `InstallType`, and moving environment handling out of `multirust-dist` and into the `Toolchain` type.

The only major new functionality here is validating the `available` flags in manifests.

This is diverging heavily from master now.

r? @Diggsey
cc @alexcrichton @aturon

I've been working on the v2 manifest generation on the server and should have that up soon. In conjunction with that I'll be testing and cleaning up the UI.

I have a big list of todo's now and intend to write them up on the issue tracker soon.

### CODE REVIEW

File: src/multirust-mock/src/clitools.rs [hunk 1]
```diff
@@ -301,12 +299,27 @@ fn build_mock_channel(channel: &str, date: &str,
...
 
-pub fn this_host_triple(channel: &str) -> String {
-    ToolchainDesc::from_str(channel).ok().map(|t| t.target_triple()).unwrap()
+pub fn this_host_triple() -> String {
+    let arch = if cfg!(target_arch = "x86") { "i686" }
+    else if cfg!(target_arch = "x86_64") { "x86_64" }
+    else { unimplemented!() };
+    let os = if cfg!(target_os = "linux") { "unknown-linux" }
+    else if cfg!(target_os = "windows") { "windows" }
+    else if cfg!(target_os = "macos") { "macos" }
```
Comment (alexcrichton) [inline src/multirust-mock/src/clitools.rs]: I may have just randomly come across this by accident, but shouldn't this be `apple-darwin` and the Windows one be `pc-windows`?

File: src/multirust-dist/tests/channel-rust-nightly-example2.toml [hunk 2]
```diff
@@ -0,0 +1,728 @@
+manifest-version = "2"
```
Comment (alexcrichton) [inline src/multirust-dist/tests/channel-rust-nightly-example2.toml]: mmm manifest goodness

File: src/multirust-mock/src/clitools.rs (hunk 1 above)
Comment (brson) [inline src/multirust-mock/src/clitools.rs]: Yep!

I should really test this harder...

**Comment by brson** [comment]
I've just been pushing my work here as I go. Now doing UI cleanup.

**Comment by brson** [comment]
I'll follow up with another pr soon.

**Pull request closed without merging**